server/fileserver.go

```diff
@@ -24,6 +25,52 @@ func getUploadHandler(fsPath string) http.Handler {
 				return
 			}
 		}()
+		path := req.URL.Path
+		filePath := filepath.Join(fsPath,req.URL.String())
+		reqType := req.Header.Get("Content-type")
+		if !strings.Contains(reqType, multipartform){
+			// copy the body of the request to the file.
+			// in this case the url will be used for the file name.
+			// example:  a/b/c.txt in the request will form a file named c.txt in a/b and its content
+			// will be the body of the request.
+
+			err := os.MkdirAll(filepath.Dir(filePath), 0755)
+			if err != nil {
+				logger.WithError(err).Error("Error creating user directories")
+				status = http.StatusInternalServerError
+				return
+			}
+			//check if the file exist. if yes delete it first.
+			// TODO: check david if its safe for us.
+			if _, err := os.Stat(filePath); !os.IsNotExist(err) {
+				err := os.Remove(filePath)
+				if err != nil {
+					logger.WithError(err).Error("Error removing existing file and replacing it")
+					status = http.StatusInternalServerError
+					return
+				}
+			}
+			out, err := os.Create(filePath)
+			if err != nil {
+				logger.WithError(err).Error("Error creating user file (raw data from body)")
+				status = http.StatusInternalServerError
+				return
+			}
+			defer func() {
+				err = out.Close()
+				if err != nil {
+					writeResponse(res, req, status, &Response{err.Error()})
+					return
+				}
+			}()
+			_, err = io.Copy(out, req.Body)
+			if err != nil {
+				logger.WithError(err).Error("Error copying the request body to file (raw data from body)")
+				status = http.StatusInternalServerError
+				return
+			}
+			return
```

Comment:
write a response here like done below for multipart form (num of bytes written, status accepted, etc...)